cpp/scripts/cleanup-doxygen.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+# Copyright (c) 2021, NVIDIA CORPORATION.
+
+# Run using output from build. i.e.
+# `./build.sh cppdocs > cppdocs.log && ./cpp/scripts/cleanup-doxygen.sh --log-file ./cppdocs.log`
+# 
+# Or pipe the output directly to the script
+# `./build.sh cppdocs | ./cpp/scripts/cleanup-doxygen.sh`
+
+DOC_LIST=()
+DO_PROCESS_LOG=0
+
+
+# Long arguments
+LONG_ARGUMENT_LIST=(
+    "file:"
+    "log-file:"
+    "all"
+)
+
+# Short arguments
+ARGUMENT_LIST=(
+    "f:"
+)
+
+# read arguments
+opts=$(getopt \
+    --longoptions "$(printf "%s," "${LONG_ARGUMENT_LIST[@]}")" \
+    --name "$(basename "$0")" \
+    --options "$(printf "%s" "${ARGUMENT_LIST[@]}")" \
+    -- "$@"
+)
+
+if [ $? != 0 ] ; then echo "Terminating..." >&2 ; exit 1 ; fi
+
+eval set -- "$opts"
+
+while true; do
+    case "$1" in
+        h)
+            show_help
+            exit 0
+            ;;
+        -f | --file )
+            shift
+            echo "Adding file $1"
+            DOC_LIST=("${DOC_LIST[@]}" "$1")
+            ;;
+        --log-file )
+            shift
+            echo "Adding log file $1"
+            # DOC_LIST=("${DOC_LIST[@]}" $(sed -nr 's/^.*?(\/home\/mdemoret\/Repos\/rapids\/cuml-dev2\/cpp\/.+?\.(cuh?|hp?p?|cpp)).*$/\1/p' $1  | awk '!seen[$0]++'))
+            INPUT_LOG_FILE="$1"
+            ;;
+        --all )
+            echo "Adding all"
+            DOC_LIST=("${DOC_LIST[@]}" $(find . -iregex '\./cpp/.*\.\(h\|hpp\|cpp\|cuh\|cu\)$'))
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+    shift
+done
+
+# Check to see if a pipe exists on stdin.
+if [ -p /dev/stdin ]; then
+    echo "Log was piped to this script!"
+
+    # STDIN_DATA=$(cat)
+
+    # echo "Input: $STDIN_DATA"
+
+    # STDIN_DOC_LIST=$(sed -nr 's/^.*?(\/home\/mdemoret\/Repos\/rapids\/cuml-dev2\/cpp\/.+?\.(cuh?|hp?p?|cpp)).*$/\1/p' | awk '!seen[$0]++')
+
+    # DOC_LIST=("${DOC_LIST[@]}" $(sed -nr 's/^.*?(\/home\/mdemoret\/Repos\/rapids\/cuml-dev2\/cpp\/.+?\.(cuh?|hp?p?|cpp)).*$/\1/p' | awk '!seen[$0]++'))
+    
+    # Just ensure INPUT_LOG_FILE is set to null
+    INPUT_LOG_FILE=
+fi
+
+if [[ -n "${INPUT_LOG_FILE+1}" ]]; then
+    echo "INPUT_LOG_FILE defined: '${INPUT_LOG_FILE}', Appending"
+    DOC_LIST=("${DOC_LIST[@]}" $(cat ${INPUT_LOG_FILE} | sed -nr 's/^.*?(\/home\/mdemoret\/Repos\/rapids\/cuml-dev2\/cpp\/.+?\.(cuh?|hp?p?|cpp)).*$/\1/p' | awk '!seen[$0]++'))
```

Comment:
folder names seem to be hard-coded?

Reply:
Ya, creating a regex from the current folder name was a pain. Since this is still a WIP, I just used the hard-coded path.